environment/src/main/java/com/sequenceiq/environment/environment/flow/creation/handler/freeipa/FreeIpaCreationHandler.java

```diff
@@ -143,13 +151,33 @@ private void createFreeIpa(Event<EnvironmentDto> environmentDtoEvent, Environmen
                 dnsV1Endpoint.addDnsZoneForSubnetIds(addDnsZoneForSubnetIdsRequest);
             }
         } else {
-            LOGGER.info("FreeIpa for environmentCrn '{}' already exists. Using this one.", environment.getResourceCrn());
+            LOGGER.info("FreeIpa for environmentCrn '{}' already exists. Using this one.", environmentDto.getResourceCrn());
             if (CREATE_IN_PROGRESS == freeIpa.get().getStatus()) {
                 awaitFreeIpaCreation(environmentDtoEvent, environmentDto);
             }
         }
     }
 
+    private void attachParentFreeIpa(EnvironmentDto environmentDto) throws Exception {
+        String parentEnvironmentCrn = environmentDto.getParentEnvironmentCrn();
+        Optional<DescribeFreeIpaResponse> freeIpa = freeIpaService.describe(parentEnvironmentCrn);
+        if (freeIpa.isPresent() && CloudPlatform.YARN.name().equalsIgnoreCase(environmentDto.getCloudPlatform())) {
```

Comment:
should we use the mock as well ? If we dont have this option for mock I am afraid that we can easily break this feature.

Reply:
Done in af20f701d8c3e87addc57f612c9b6fc642d0daf5